internal/worker/worker.go

```diff
@@ -18,43 +20,57 @@ const (
 )
 
 func GenerateMosaic(
+	ctx context.Context,
 	m mosaic.Mosaic,
 	cfg *config.Config,
 	logger *zap.SugaredLogger,
 	locker locking.Locker,
 	cmdExecutor mosaic.Command,
-	runningProcesses map[string]bool,
+	runningProcesses *sync.Map,
 	stg storage.Storage,
 ) error {
-	_, exists := runningProcesses[m.Name]
-	if exists {
+	if _, exists := runningProcesses.Load(m.Name); exists {
 		return nil
 	}
 
+	runningProcesses.Store(m.Name, true)
+
+	defer runningProcesses.Delete(m.Name)
+
```

Comment:
you'll remove all this code right? once this is managed by the system you won't need to bother, righ?

Reply:
You are right!

Reply:
Removed.